tests/components/onkyo/test_media_player.py

```diff
@@ -0,0 +1,272 @@
+"""Test Onkyo media player platform."""
+
+from collections.abc import AsyncGenerator
+from unittest.mock import AsyncMock, patch
+
+from aioonkyo import Instruction, Zone, command
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.media_player import (
+    ATTR_INPUT_SOURCE,
+    ATTR_MEDIA_CONTENT_ID,
+    ATTR_MEDIA_CONTENT_TYPE,
+    ATTR_MEDIA_VOLUME_LEVEL,
+    ATTR_MEDIA_VOLUME_MUTED,
+    ATTR_SOUND_MODE,
+    DOMAIN as MEDIA_PLAYER_DOMAIN,
+    SERVICE_PLAY_MEDIA,
+    SERVICE_SELECT_SOUND_MODE,
+    SERVICE_SELECT_SOURCE,
+)
+from homeassistant.components.onkyo.services import (
+    ATTR_HDMI_OUTPUT,
+    SERVICE_SELECT_HDMI_OUTPUT,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    SERVICE_VOLUME_DOWN,
+    SERVICE_VOLUME_MUTE,
+    SERVICE_VOLUME_SET,
+    SERVICE_VOLUME_UP,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ServiceValidationError
+from homeassistant.helpers import entity_registry as er
+
+from . import setup_integration
+
+from tests.common import MockConfigEntry, snapshot_platform
+
+ENTITY_ID = "media_player.tx_nr7100"
+ENTITY_ID_ZONE_2 = "media_player.tx_nr7100_zone_2"
+ENTITY_ID_ZONE_3 = "media_player.tx_nr7100_zone_3"
+
+
+@pytest.fixture(autouse=True)
+async def auto_setup_integration(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_receiver: AsyncMock,
+) -> AsyncGenerator[None]:
+    """Auto setup integration."""
+    with (
+        patch.multiple(
+            "homeassistant.components.onkyo.media_player",
+            AUDIO_VIDEO_INFORMATION_UPDATE_WAIT_TIME=0,
+        ),
+        patch("homeassistant.components.onkyo.PLATFORMS", [Platform.MEDIA_PLAYER]),
+    ):
+        await setup_integration(hass, mock_config_entry)
+        yield
+
+
+async def test_entities(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    mock_config_entry: MockConfigEntry,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test entities."""
+    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
+
+
+async def test_turn_on(hass: HomeAssistant, writes: list[Instruction]) -> None:
+    """Test turn on."""
+    writes.clear()
```

Comment:
if we do something cool like this, let's make it even cooler and have it auto clear